### PR TITLE
add Non444ChromaSubsampling check

### PR DESCRIPTION
**Pull request by mo271**

### CODE REVIEW

File: jxl/src/headers/frame_header.rs [hunk 1]
```diff
@@ -417,6 +420,26 @@ pub struct FrameHeader {
     #[default(if frame_height == 0 { nonserialized.img_height } else { frame_height })]
     #[condition(false)]
     pub height: u32,
+
+    #[coder(Bits(0))]
+    #[default({
```
Comment (veluca93) [inline jxl/src/headers/frame_header.rs:425]: Make that into a free function? It's a fair amount of code for an attribute, and effectively duplicated

Comment (mo271) [reply jxl/src/headers/frame_header.rs]: like this?

File: jxl/src/headers/frame_header.rs [hunk 2]
```diff
@@ -430,22 +451,13 @@ impl FrameHeader {
...
-        let mut maxvs = 0;
-        for ch in self.jpeg_upsampling {
-            maxhs = maxhs.max(H_SHIFT[ch as usize]);
-            maxvs = maxvs.max(V_SHIFT[ch as usize]);
-        }
         let xsize = self.width as usize;
         let ysize = self.height as usize;
-        let xsize_blocks = xsize.div_ceil(BLOCK_DIM << maxhs) << maxhs;
-        let ysize_blocks = ysize.div_ceil(BLOCK_DIM << maxvs) << maxvs;
+        let xsize_blocks = xsize.div_ceil(BLOCK_DIM << self.maxhs) << self.maxhs;
```
Comment (veluca93) [inline jxl/src/headers/frame_header.rs:459]: For later: might now make sense to move this logic to `num_groups`/`num_dc_groups` and drop `group_counts`

File: jxl/src/headers/frame_header.rs [hunk 3]
```diff
@@ -256,6 +256,17 @@ pub struct FrameHeaderNonserialized {
     pub img_height: u32,
 }
 
+const H_SHIFT: [usize; 4] = [0, 1, 1, 0];
+const V_SHIFT: [usize; 4] = [0, 1, 0, 1];
+
+fn compute_max(jpeg_upsampling: &[u32], shift_table: &[usize]) -> u32 {
```
Comment (veluca93) [inline jxl/src/headers/frame_header.rs:262]: nit: compute_jpeg_shift?

File: jxl/src/headers/frame_header.rs (hunk 2 above)
Comment (mo271) [reply jxl/src/headers/frame_header.rs:459]: done, see #50

**Pull request merged**